packages/ui/src/components/pagination/pagination.tsx

```diff
@@ -1,151 +1,238 @@
       </PaginationPrimitive.Item>
     )
   }
 
   // Add ellipsis if needed
-  if (currentPage < totalPages - siblings - 1) {
+  if (rightBound < totalPages - 1) {
     items.push(
-      <PaginationPrimitive.Item key="end-ellipsis">
+      <PaginationPrimitive.Item key="end-ellipsis" className="cn-pagination-pages">
         <PaginationPrimitive.Ellipsis />
       </PaginationPrimitive.Item>
     )
   }
 
-  // Always show the last page
-  items.push(
-    <PaginationPrimitive.Item key={totalPages}>
-      <PaginationPrimitive.Link isFullRounded onClick={goToPage(totalPages)} isActive={currentPage === totalPages}>
-        {totalPages}
-      </PaginationPrimitive.Link>
-    </PaginationPrimitive.Item>
-  )
+  // Always show the last page if it's different from the first page
+  if (totalPages > 1) {
+    items.push(
+      <PaginationPrimitive.Item key={totalPages} className="cn-pagination-pages">
+        <PaginationPrimitive.Link
+          href={getPageLink?.(totalPages)}
+          onClick={goToPage?.(totalPages)}
+          isActive={currentPage === totalPages}
+        >
+          {totalPages}
+        </PaginationPrimitive.Link>
+      </PaginationPrimitive.Item>
+    )
+  }
 
   return <>{items}</>
 }
 
-export interface PaginationProps {
-  currentPage: number
-  goToPage: (pageNum: number) => void
-  totalPages?: number
-  nextPage?: number
-  previousPage?: number
+interface PaginationBaseProps {
   className?: string
   t: TFunction
+  truncateLimit?: number
 }
 
+type DeterminatePaginationNavProps =
+  | { goToPage: (page: number) => void; getPageLink?: never }
+  | { goToPage?: never; getPageLink: (page: number) => string }
+
+type IndeterminatePaginationNavProps =
+  | { onPrevious: () => void; onNext: () => void; getPrevPageLink?: never; getNextPageLink?: never }
+  | { onPrevious?: never; onNext?: never; getPrevPageLink: () => string; getNextPageLink: () => string }
+
+type DeterminatePaginationProps = PaginationBaseProps &
+  DeterminatePaginationNavProps & {
+    totalItems: number
+    pageSize: number
+    currentPage: number
+    hidePageNumbers?: boolean
+    indeterminate?: false
+
+    hasPrevious?: never
+    hasNext?: never
+    getPrevPageLink?: never
+    getNextPageLink?: never
+    onPrevious?: never
+    onNext?: never
+  }
+
+type IndeterminatePaginationProps = PaginationBaseProps &
+  IndeterminatePaginationNavProps & {
+    hasPrevious?: boolean
+    hasNext?: boolean
+    indeterminate: true
+
+    goToPage?: never
+    getPageLink?: never
+    totalItems?: never
+    pageSize?: never
+    currentPage?: never
+    hidePageNumbers?: never
+  }
+
+export type PaginationProps = DeterminatePaginationProps | IndeterminatePaginationProps
+
 export const Pagination: FC<PaginationProps> = ({
-  totalPages,
-  nextPage,
-  previousPage,
+  totalItems,
+  pageSize,
   currentPage,
   goToPage,
+  getPageLink,
+  hasNext,
+  hasPrevious,
   className,
-  t
+  t,
+  truncateLimit = 4,
```

Comment:
We had agreed to remove this as a prop to ensure consistency and setting as `5` since it was most correct when detached from both sides 🤔